### ISSUE TITLE
Lens.modify loses reference to "this" when used as standalone function

### ISSUE THREAD

**Issue by VincentQuillien** [gcanti/monocle-ts#128]
### Current Behavior

When `Lens.modify` is used as a function it loses the reference to `this`

```ts
import { Lens } from 'monocle-ts';

type Test = {
    a: number;
};

const lens = Lens.fromProp<Test>()('a');

const test: Test = { a: 42 };

const add42 = (n: number) => n + 42;

lens.modify(add42)(test) // <--- succeeds 

const { modify } = lens;
modify(add42)(test); // <--- fails

```

![Capture d’écran 2020-06-17 à 17 54 13](https://user-images.githubusercontent.com/22659282/84920606-af09fc80-b0c3-11ea-87cb-af02178f4d92.png)

### Expected behavior

It would be nice to be able to use functional composition with `Lens.modify` :

```ts
import { pipe } from 'fp-ts/lib/function'

const add42ToA = pipe(add42, lens.modify)
```

### Your environment

Which versions of monocle-ts are affected by this issue? Did this work in previous versions of monocle-ts?

| Software   | Version(s) |
| ---------- | ---------- |
| monocle-ts |    2.2.0        |
| fp-ts      |      2.6.6    |
| TypeScript |     3.9.4       |

**Comment by VincentQuillien** [comment]
Looks like it will be resolved with 2.3's modules, looking forward to it !

**Comment by gcanti** [comment]
@VincentQuillien keep in mind that those modules will be released as experimental for some time, though I don't expect drastic changes (I'm quite satisfied by the APIs). I just wanted to make sure you were aware of that.
